### PR TITLE
Remove About page and add Releases page

### PR DESCRIPTION
**Pull request by flands**
- Addresses #379 
- Addresses #170

### CODE REVIEW

File: content/en/releases/_index.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+Our goal is to provide a generally available, production quality release for
+the tracing data source across most OpenTelemetry components in the first half
+of 2021. Several components have already reached this milestone!
+
+We expect metrics to reach the same status in the second half of 2021 and are
+targeting logs in 2022.
+
+{{% /blocks/cover %}}
+
+{{% blocks/section type="section" color="primary" %}}
```
Comment (austinlparker) [inline content/en/releases/_index.md:19]: Maybe this should be secondary color? The blue on blue link text is hard to read.

Comment (flands) [reply content/en/releases/_index.md:19]: Yeah, we had the same issue on https://opentelemetry.io/about/. I opened #384 to track this and should have linked here.

Given #383 and #303 I actually have a PR on top of this one that switches to the layout that community/registry use while adding a dark middle section so links are visible.

Comment (flands) [reply content/en/releases/_index.md:19]: See #386 -- perhaps we could merge this then land the supplemental PR. @austinlparker thoughts?

Comment (austinlparker) [reply content/en/releases/_index.md:19]: yeah, that works for me. #386 looks solid.

**Review by austinlparker — APPROVED** [review]

**Pull request merged**